fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -4268,29 +4270,47 @@ class FMCMainDisplay extends BaseAirliners {
     }
 
     checkEFOBBelowMin() {
+        if (this._fuelPredDone) {
         if (!this._minDestFobEntered) {
             this.tryUpdateMinDestFob();
         }
-        const EFOBBelMin = this.isAnEngineOn() ? this.getDestEFOB(true) : this.getDestEFOB(false);
 
-        if (EFOBBelMin < this._minDestFob) {
-            if (this.isAnEngineOn()) {
-                setTimeout(() => {
-                    this.addMessageToQueue(NXSystemMessages.destEfobBelowMin, () => {
-                        return this._EfobBelowMinClr === true;
-                    }, () => {
-                        this._EfobBelowMinClr = true;
-                    });
-                }, 180000);
+        if (this._minDestFob) {
+            // round & only use 100kgs precision since thats how it is displayed in fuel pred
+            const destEfob = Math.round(this.getDestEFOB(this.isAnEngineOn()) * 10) / 10;
+            const roundedMinDestFob =  Math.round(this._minDestFob * 10) / 10;
+            if (!this._isBelowMinDestFob) {
+                if (destEfob < roundedMinDestFob) {
+                    this._isBelowMinDestFob = true;
+                    // TODO should be in flight only and if fuel is below min dest efob for 2 minutes
+                    if (this.isAnEngineOn()) {
+                        setTimeout(() => {
+                            this.addMessageToQueue(NXSystemMessages.destEfobBelowMin, () => {
+                                return this._EfobBelowMinClr === true;
+                            }, () => {
+                                this._EfobBelowMinClr = true;
+                            });
+                        }, 120000);
```

Comment:
I'm not quite sure this bit of logic works as intended. I assume you only want to show the message if the condition `destEfob < roundedMinDestFob` is met for two minutes, and reset the timer if at any point the condition is no longer true. Whereas at the moment, you always show the message after a two minute delay even if the condition is only momentarily true.

Reply:
Yes you are completely right. Now that I think about it (its been a while ago 😅 ) I did not touch the original logic of this scratchpad message hence why the TODO comment as I'd rather do it in a separate PR preferably in typescript as some of the others fms-messages if it is possible to access the fuel predictions through there. Nonetheless, I could fix it in this PR if you wish. Formatting issues should be fixed too.

Reply:
Alright, a separate PR is fine by me 🙂